src/main/scala/diplomacy/SRAM.scala

```diff
@@ -34,16 +32,43 @@ abstract class DiplomaticSRAM(
   def mask: List[Boolean] = bigBits(address.mask >> log2Ceil(beatBytes))
 
   // Use single-ported memory with byte-write enable
-  def makeSinglePortedByteWriteSeqMem(size: BigInt, lanes: Int = beatBytes, bits: Int = 8) = {
+  def makeSinglePortedByteWriteSeqMem(
+    size: BigInt,
+    lanes: Int = beatBytes,
+    bits: Int = 8,
+    busProtocol: Option[OMProtocol],
+    dataECC: Option[OMECC] = None,
+    hasAtomics: Option[Boolean] = None) = {
```

Comment:
I think you should return `omSRAM` from this function and add it to logical tree at the call site, instead of having to pass all these extra arguments in to this function that are only used to make the logical tree node